internal/transport/grpchttp2/http2bridge_test.go

```diff
@@ -0,0 +1,625 @@
+	fields := []hpack.HeaderField{
+		{Name: "foo", Value: "bar"},
+		{Name: "baz", Value: "qux"},
+	}
+
+	c := &testConn{}
+	enc := hpack.NewEncoder(c)
+	for _, field := range fields {
+		enc.WriteField(field)
+	}
+	half1 := c.wbuf[0 : len(c.wbuf)/2]
+	half2 := c.wbuf[len(c.wbuf)/2:]
+
+	// Writing a Headers Frame with half of the encoded headers
+	c.rbuf = append(c.rbuf, 0, 0, byte(len(half1)), byte(FrameTypeHeaders), 0)
+	c.rbuf = appendUint32(c.rbuf, 1)
+	c.rbuf = append(c.rbuf, half1...)
+
+	// Writing a Continuation Frame with the other half of the encoded headers
+	// to test merging.
+	c.rbuf = append(c.rbuf, 0, 0, byte(len(half2)), byte(FrameTypeContinuation), byte(FlagContinuationEndHeaders))
+	c.rbuf = appendUint32(c.rbuf, 1)
+	// Copy data written by the encoder into the reading buf
+	c.rbuf = append(c.rbuf, half2...)
+
+	f := NewFramerBridge(c, c, 0, nil)
+	fr, err := f.ReadFrame()
+	if err != nil {
+		t.Fatalf("ReadFrame(): %v", err)
+	}
+
+	mf, ok := fr.(*MetaHeadersFrame)
+	if !ok {
+		t.Errorf("ReadFrame(): Type: expected MetaHeadersFrame, got %T", fr)
+	}
+	if len(mf.Fields) != 2 {
+		t.Errorf("ReadFrame(): Fields: got %d, want %d", len(mf.Fields), 1)
+	}
+	for i, field := range fields {
+		if field.Name != mf.Fields[i].Name {
+			t.Errorf("ReadFrame(): Fields[%d].Name: got %q, want %q", i, mf.Fields[i].Name, field.Name)
+		}
+		if field.Value != mf.Fields[i].Value {
+			t.Errorf("ReadFrame(): Fields[%d].Value: got %q, want %q", i, mf.Fields[i].Value, field.Value)
+		}
+	}
+}
+
+// Tests and verifies that the framer correctly reads an unknown frame Frame.
+func (s) TestBridge_ReadFrame_UnknownFrame(t *testing.T) {
+	c := &testConn{}
+	wantData := "test data"
+
+	// Writing an unknown frame to the reading buf
+	c.rbuf = append(c.rbuf, 0, 0, byte(len(wantData)), 0xa, 0)
+	c.rbuf = appendUint32(c.rbuf, 1)
+	c.rbuf = append(c.rbuf, []byte(wantData)...)
+
+	f := NewFramerBridge(c, c, 0, nil)
+	fr, err := f.ReadFrame()
+	if err != nil {
+		t.Errorf("ReadFrame(): %s", err)
+	}
+
+	wantHdr := &FrameHeader{
+		Size:     uint32(len(wantData)),
+		Type:     0xa,
+		Flags:    0,
+		StreamID: 1,
+	}
+	if diff := cmp.Diff(fr.Header(), wantHdr); diff != "" {
+		t.Errorf("ReadFrame(): %s", diff)
+	}
+
+	uf, ok := fr.(*UnknownFrame)
+	if !ok {
+		t.Errorf("ReadFrame(): Type: expected UnknownFrame, got %T", f)
+	}
+	if string(uf.Payload) != wantData {
+		t.Errorf("ReadFrame(): Payload: got %q, want %q", uf.Payload, wantData)
+	}
+	uf.Free()
+}
+
+// Tests and verifies that a Data Frame is correctly written.
+func (s) TestBridge_WriteData(t *testing.T) {
+	c := &testConn{}
+	testBuf := [][]byte{[]byte("test"), []byte(" data")}
+	f := NewFramerBridge(c, c, 0, nil)
+	f.WriteData(1, false, testBuf...)
+
+	wantData := "test data"
+	wantHdr := &FrameHeader{
+		Size:     uint32(len(wantData)),
+		Type:     FrameTypeData,
+		Flags:    0,
+		StreamID: 1,
+	}
+	if errs := checkWrittenHeader(c.wbuf[0:9], wantHdr); len(errs) > 0 {
```

Comment:
I somehow remember talking about it in the last pass, but now I'm not able to find that conversation..

I dont want to block on this, but I would prefer if this helper was called parseWrittenHeader which returns FrameHeader, and call cmp.Diff (like in the other tests) here for assertion

Reply:
Done.